src/components/BrowserFilter/FilterRow.react.js

```diff
@@ -49,7 +49,18 @@ function compareValue(info, value, onChangeCompareTo, active) {
     case 'Boolean':
       return <ChromeDropdown color={active ? 'blue' : 'purple'} value={value ? 'True' : 'False'} options={['True', 'False']} onChange={(val) => onChangeCompareTo(val === 'True')} />;
     case 'Number':
-      return <input type='text' value={value} onChange={(e) => onChangeCompareTo(validateNumeric(e.target.value) ? parseFloat(e.target.value) : (parseFloat(value) || ''))} />;
+      return (
+        <input
+          type='text'
+          value={value}
+          onChange={(e) => {
+            let val = value;
+            if (!e.target.value.length) val = '';
```

Comment:
Please don't use `if` and `else` without brackets. This might lead to bugs. And already did in the history of programming many times :) Also, don't keep the `val` assignment in the same line.